src/inc/regdisp.h

```diff
@@ -396,7 +396,7 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 #elif defined(_TARGET_X86_) // _TARGET_ARM_
     for (int i = 0; i < 7; i++)
     {
-        *(&pRD->ctxPtrsOne.Esi + i) = (&pctx->Esi + i);
+        *(&pRD->ctxPtrsOne.Esi + i) = (&pRD->pCurrentContext->Esi + i);
```

Comment:
@parjong the semantics of the FillRegDisplay is to fill the REGDISPLAY from the passed in pctx and using the registers stored in the pctx as the initial locations where the context pointers point. Your change would change that semantics so that the context pointers would point into the context pointed to by the pCurrentContext instead. I would prefer keeping the semantics the same as for other target architectures. 
The *pCurrentContext is initialized with the *pctx values at line 352, so the values that the context pointers point to are the same with or without your change. The only difference is their location.
If I understand what you are saying correctly, the InitializeCurrentContextForCrawlFrame is called after the FillRegDisplay and patches only the pCurrentContext, which causes a trouble later when the GetRegdisplayFP is called, since it gets the value using the context pointers.
I think the fix should rather be in the InitializeCurrentContextForCrawlFrame to let it update the pCurrentContextPointers to point to the registers in the pDispatcherContext->CurrentNonVolatileContextRecord.